### PR TITLE
New OAS Renderer

### PR DESCRIPTION
**Pull request by mheap**
This is a *huge* PR and likely has multiple refactoring opportunities.

To see it rendered, use https://nexmo-oas-redesign.herokuapp.com/

Some specs might not look quite right until outstanding PRs are merged on the API Specification repo (e.g. Voice)

### CODE REVIEW

File: lib/nexmo/oas/renderer/presenters/request_body_raw.rb [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        class RequestBodyRaw
+          def initialize(parameters, options = {}, endpoint = nil)
+            @parameters = parameters
+            @options = options
+            @endpoint = endpoint
+          end
+
+          def to_format(format)
+            return to_urlencoded if format == "application/x-www-form-urlencoded"
+            return to_json
```
Comment (fabianrbz) [inline lib/nexmo/oas/renderer/presenters/request_body_raw.rb:14]: the `return` here is not necessary

File: lib/nexmo/oas/renderer/presenters/request_body_raw.rb [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          end
+
+          def to_urlencoded
+
```
Comment (fabianrbz) [inline lib/nexmo/oas/renderer/presenters/request_body_raw.rb:18]: empty line ✂️

File: lib/nexmo/oas/renderer/presenters/request_body_raw.rb [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            example = ""
+            body = URI.encode_www_form(generate_request)
+            if @endpoint
+              servers = @endpoint.path.servers ? @endpoint.path.servers : @endpoint.definition.servers
```
Comment (fabianrbz) [inline lib/nexmo/oas/renderer/presenters/request_body_raw.rb:22]: you can also do `@endpoint.path.servers || @endpoint.definition.servers`

**Comment by mheap** [comment]
All comments addressed by adding Rubocop to the build 👍

**Pull request merged**